rclpy/src/rclpy/_rclpy.c

```diff
@@ -1599,6 +1599,72 @@ rclpy_publish(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Publish a serialized message
+/**
+ * Raises ValueError if pypublisher is not a publisher capsule
+ * Raises RuntimeError if the message cannot be published
+ *
+ * \param[in] pypublisher Capsule pointing to the publisher
+ * \param[in] pymsg serialized message to send
+ * \return NULL
+ */
+
+static PyObject *
+rclpy_publish_raw(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pypublisher;
+  PyBytesObject * pymsg;
+  char * message;
+  Py_ssize_t pymsg_size;
+
+  if (!PyArg_ParseTuple(args, "OO", &pypublisher, &pymsg)) {
```

Comment:
If I understand the [documentation](https://docs.python.org/3.6/c-api/arg.html#strings-and-buffers), I think we should use `S` for objects of type `bytes`:
```suggestion
  if (!PyArg_ParseTuple(args, "OS", &pypublisher, &pymsg)) {
```

Reply:
Maybe this is the reason for the test failure?

Reply:
That could be it. I was previously using `PyBytes_AsString` and `PyBytes_Size` to fetch the buffer contents and size. Once I switched to `PyBytes_AsStringAndSize` I stopped receiving that error in the test